### PR TITLE
LayeredCircuit + Summon

### PR DESCRIPTION
**Pull request by voltrevo**
Adds `LayeredCircuit` which implements `::from_bristol` and `::from_summon`. Basically a boolean circuit but separated into layers that can be evaluated in parallel.

Implements `eval(inputs)` which works with `bool` and `FheBool`, tested in `layered_circuit_demo.rs` and `summon_demo.rs`.

### CODE REVIEW

**Comment by Janmajayamall** [comment]
@voltrevo would it be possible merge to `rewrite` branch instead? 

Also, would it be possible to also paste the file string in the example?

**Comment by voltrevo** [comment]
@Janmajayamall Done and done.

File: frontend/src/layered_circuit.rs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+        );
+
+        let inputs = io_labels(
+            &bristol_circuit.info.input_name_to_wire_index,
+            bristol_circuit.io_widths.0.clone(),
+        );
+
+        let outputs = io_labels(
+            &bristol_circuit.info.output_name_to_wire_index,
+            bristol_circuit.io_widths.1.clone(),
```
Comment (Janmajayamall) [inline frontend/src/layered_circuit.rs:142]: I see BristolCircuit has io_width. But it isn't clear what exactly is io_width. 

@voltrevo possible to add small explanation here ?

Comment (Janmajayamall) [reply frontend/src/layered_circuit.rs:142]: does `width` mean the bit base? For ex, if width is 1 then values are binary. If width is 2 then values are in range [0, 2^2)...etc.

File: frontend/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+summon_compiler = { git = "https://github.com/voltrevo/summon", rev = "0eb8ff2" }
+summon_cli = { git = "https://github.com/voltrevo/summon", rev = "0eb8ff2" }
+boolify = { git = "https://github.com/voltrevo/boolify", rev = "204bf68" }
+serde = { workspace = true, features = ["derive"] }
+serde_json = { workspace = true }
+rayon = { workspace = true }
+
+[dev-dependencies]
+rand = { workspace = true }
+phantom-zone-evaluator = { path = "../evaluator" }
```
Comment (Janmajayamall) [inline frontend/Cargo.toml:17]: both examples require 'dev' of "phantom-zone/crypto". Probably add the following to the toml file.

```
[features]
default = []
dev = ["phantom-zone-evaluator/dev"]

[[example]]
name = "layered_circuit_demo"
required-features = ["dev"]

[[example]]
name = "summon_demo"
required-features = ["dev"]
```

File: frontend/examples/summon_demo.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+            Err("File not found".to_string())
+        },
+    );
+
+    println!("{:?}", layered_circuit);
+
+    let mut rng = StdLweRng::from_entropy();
+    let sk = LweSecretKey::sample(PARAM.ring_size, PARAM.sk_distribution, &mut rng);
+    let evaluator = Evaluator::sample(PARAM, &sk, &mut rng);
```
Comment (Janmajayamall) [inline frontend/examples/summon_demo.rs:76]: supply `&sk` as second argument to `sample` fn?

**Pull request merged**